libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/NavigationRouteCallback.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.navigation.route.offboard.router
+
+import com.mapbox.api.directions.v5.models.DirectionsResponse
+import com.mapbox.navigation.utils.time.ElapsedTime
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class NavigationRouteCallback(
+    private val listener: NavigationRouteEventListener,
+    private val callback: Callback<DirectionsResponse>
```

Comment:
is there a reason why this class accepts `Callback<DirectionsResponse>` as a constructor parameter and at the same time extends it?

Reply:
looks like it's needed for different aims, inner callback is used to delegate response